gkeys/gkeys/email_reminders.py

```diff
@@ -67,22 +70,25 @@ def _generate_template(self, template_path, keyprints, specprint):
 
     def _is_expiring(self):
         is_expiring = False
-        if self.pub.days < 180 and self.pub.days > -180:
+        days_limit = self.config.get_key('days_limit')
+        if self.pub.days < days_limit and self.pub.days > days_limit*(-1):
             is_expiring = True
         else:
             for subkey in self.sub:
-                if subkey.days < 180 and subkey.days > -180:
+                if subkey.days < days_limit and subkey.days > days_limit*(-1):
                     is_expiring = True
                     break
         return is_expiring
 
     def email_reminder(self, keyprints, specprint):
         is_expiring = self._is_expiring()
+        config_dir = self.config.get_key('userconfigdir')
+        template_dir = os.path.join(config_dir + '/templates')
+        self.logger.debug("User template path is:" + template_dir)
         if self.uid and is_expiring:
             template_path = '/usr/share/gkeys/templates/email_template'
-            if os.path.exists('/home/aeroniero/.config/gkeys/templates/email_template'):
-                template_path = '/home/aeroniero/.config/gkeys/templates/email_template'
-            elif os.path.exists('/var/lib/gentoo/gkeys/templates/email_template'):
-                template_path = '/var/lib/gentoo/gkeys/templates/email_template'
+            if os.path.exists(template_dir + '/email_template'):
+                template_path = template_dir + '/email_template'
```

Comment:
you are still hard-coding the template to use, that too should be a config value.  Also the class should be initialized with the template to use.


Further...  Like I said in my first comments...  Thsi Emailer should be for email handling, not checking to see if it should make the email if it is expiring within so many days...  All that is already part of the scpec-check action.  the spec-check action code should only run the emailer class instance for keys that meet the spec-check email option requirement.  Emailer class is generic in that it could care less what template it uses and what data it gets to fill teh template with other than the data meets the template requirements and it has a valid email address to send it to.